### PR TITLE
Add support for DirectWrite font rendering on Windows

### LINKED ISSUE #13371: Implement font rendering on Windows (MSVC)
The dummy implementation in `components/gfx/platform/mod.rs` should be replaced by a real one:

``` rust
#[cfg(all(target_os = "windows", target_env = "msvc"))]
pub use platform::dummy::{font, font_context, font_list, font_template};
```

Currently, the lack of a proper implementation causes Servo to show a blank screen upon being launched (see https://github.com/servo/servo/issues/13257).

### PR DESCRIPTION
**Pull request by vvuk**
This PR adds support for DirectWrite text rendering using a thin wrapper for dwrite, as well as similar functionality added to WebRender.  In doing so it removes the FreeType dep from the Windows build.

There is still work to be done here, but this is a good starting point for future cleanup and fixes.

---

- [X] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [X] These changes fix #13371 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because they are already tested by other tests

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14153)

### CODE REVIEW

**Comment by vvuk** [comment]
Ugh, ignore the ipc-channel 0.6 bump commit -- I had to pull that in in order to test.  I'll rebase without it once that lands.

File: components/gfx/platform/windows/font_template.rs [hunk 1]
```diff
@@ -2,88 +2,90 @@
...
+            panic!("Can't create fonts yet");
+        }
+
+        let descriptor = descriptor_from_atom(&self.identifier);
+        Some(descriptor)
+    }
+}
+
+#[derive(Debug)]
+pub struct CachedDWFont(Mutex<HashMap<Au, Font>>);
```
Comment (glennw) [inline components/gfx/platform/windows/font_template.rs]: Is this actually used anywhere?

File: components/gfx/platform/windows/font_context.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use heapsize::HeapSizeOf;
+
+#[derive(Clone, Debug)]
+pub struct FontContextHandle {
+    ctx: ()
```
Comment (glennw) [inline components/gfx/platform/windows/font_context.rs]: Can this just be an empty struct (i.e. pub struct FontContextHandle;) ?

**Comment by glennw** [comment]
@vvuk r=me on the commits from you. If you (or someone else) wants to r= on my commits here, we can get this merged!

**Comment by vvuk** [comment]
@glennw r=me on your commits!

**Comment by glennw** [comment]
@bors-servo retry
 * Looks similar to #13133

**Pull request merged**